src/VisualStudio/TestUtilities/Window/InteractiveWindow.cs

```diff
@@ -0,0 +1,147 @@
+        private IntegrationHost _host;
+        private InteractiveWindowWrapper _interactiveWindowWrapper;
+
+        internal static InteractiveWindow CreateCSharpInteractiveWindow(IntegrationHost host)
+            => new InteractiveWindow(host, DteCSharpViewCommand, DteCSharpWindowTitle);
+
+        internal InteractiveWindow(IntegrationHost host, string dteViewCommand, string dteWindowTitle)
+        {
+            _host = host;
+            _dteViewCommand = dteViewCommand;
+
+            // We have to show the window at least once to ensure the interactive service is loaded.
+            Show(waitForPrompt: false);
+
+            var dteWindow = _host.LocateDteWindow(dteWindowTitle);
+            dteWindow.Close();
+
+            _interactiveWindowWrapper = _host.ExecuteOnHostProcess<InteractiveWindowWrapper>(typeof(RemotingHelper), nameof(RemotingHelper.CreateCSharpInteractiveWindowWrapper), (BindingFlags.Public | BindingFlags.Static));
+        }
+
+        public string LastReplOutput
+        {
+            get
+            {
+                var replText = ReplTextWithoutPrompt;
+                int lastPromptIndex = replText.LastIndexOf(ReplPromptText);
+
+                replText = replText.Substring(lastPromptIndex, (replText.Length - lastPromptIndex));
+                int lastSubmissionIndex = replText.LastIndexOf(ReplSubmissionText);
+
+                if (lastSubmissionIndex > 0)
+                {
+                    replText = replText.Substring(lastSubmissionIndex, (replText.Length - lastSubmissionIndex));
+                }
+                else if (!replText.StartsWith(ReplPromptText))
+                {
+                    return replText;
+                }
+
+                int firstNewLineIndex = replText.IndexOf(Environment.NewLine);
+
+                if (firstNewLineIndex <= 0)
+                {
+                    return replText;
+                }
+
+                firstNewLineIndex += Environment.NewLine.Length;
+                return replText.Substring(firstNewLineIndex, (replText.Length - firstNewLineIndex));
+            }
+        }
+
+        public string ReplText
+            => _interactiveWindowWrapper.CurrentSnapshotText;
+
+        public string ReplTextWithoutPrompt
+        {
+            get
+            {
+                var replText = ReplText;
+
+                // find last prompt and remove
+                int lastPromptIndex = replText.LastIndexOf(ReplPromptText);
+
+                if (lastPromptIndex > 0)
+                {
+                    replText = replText.Substring(0, lastPromptIndex);
+                }
+
+                // it's possible for the editor text to contain a trailing newline, remove it
+                return replText.EndsWith(Environment.NewLine) ? replText.Substring(0, (replText.Length - Environment.NewLine.Length)) : replText;
+            }
+        }
+
+        public bool CheckLastReplOutputEndsWith(string expectedText)
```

Comment:
I'd prefer `DoesLast...` instead of `CheckLast...` because to me "Check" implies that an assertion/error/whatever will be thrown if it doesn't meet the criteria, but I'm open to be persuaded here.